javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -82,6 +83,9 @@ public static Instrumentation getInstrumentation() {
     BootstrapPackagePrefixesHolder.setBoostrapPackagePrefixes(loadBootstrapPackagePrefixes());
     // this needs to be done as early as possible - before the first Config.get() call
     ConfigInitializer.initialize();
+    // ensure java.lang.reflect.Proxy is loaded, as transformation code uses it internally
+    // loading it after bytebuddy transformer is set up can cause a ClassCircularityError
+    Proxy.class.getName();
```

Comment:
do you have a stack trace handy for the `ClassCircularityError`?

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/runs/2447546557
https://github.com/open-telemetry/opentelemetry-java-instrumentation/runs/2447546469

Reply:
thx for tracking this down! looks like I introduced the problem in #2773